packages/medusa/src/commands/db/create.ts

```diff
@@ -0,0 +1,117 @@
+import slugify from "slugify"
+import { basename } from "path"
+import input from "@inquirer/input"
+import { logger } from "@medusajs/framework"
+import {
+  createDb,
+  dbExists,
+  EnvEditor,
+  createClient,
+  parseConnectionString,
+} from "@medusajs/utils"
+
+const main = async function ({ directory, interactive, db }) {
+  let dbName = db
+
+  /**
+   * Loading the ".env" file in editor mode so that
+   * we can read values from it and update its
+   * contents.
+   */
+  const envEditor = new EnvEditor(directory)
+  await envEditor.load()
+
+  /**
+   * Ensure the "DATABASE_URL" is defined before we attempt to
+   * create the database.
+   *
+   * Also we will discard the database name from the connection
+   * string because the mentioned database might not exist
+   */
+  const dbConnectionString = envEditor.get("DATABASE_URL")
+  if (!dbConnectionString) {
+    logger.error(
+      `Missing "DATABASE_URL" inside the .env file. The value is required to connect to the PostgreSQL server`
+    )
+    process.exitCode = 1
+    return
+  }
+
+  /**
+   * Use default value + prompt only when the dbName is not
+   * provided via a flag
+   */
+  if (!dbName) {
+    const defaultValue =
+      envEditor.get("DB_NAME") ?? `medusa-${slugify(basename(directory))}`
+    if (interactive) {
+      try {
+        dbName = await input({
+          message: "Enter the database name",
+          default: defaultValue,
+          required: true,
+        })
+      } catch (error) {
+        if (error.name === "ExitPromptError") {
+          process.exit()
+        }
+        throw error
+      }
+    } else {
+      dbName = defaultValue
+    }
+  }
```

Comment:
```suggestion
let dbName = db ?? envEditor.get("DB_NAME") ?? `medusa-${slugify(basename(directory))}`

if (interactive && !db) {
  try {
    dbName = await input({
      message: "Enter the database name",
      default: defaultValue,
      required: true,
    })
  } catch (error) {
    if (error.name === "ExitPromptError") {
      process.exit()
    }
    throw error
  }
}
```

Reply:
this is just a suggestion, I found it more easy to read wdyt? feel free to discard :)

Reply:
I think I find the existing flow of code a little better especially the way I read it. But again, I understand its a subjective thing.